### PR TITLE
Initial work on performance

### PR DESCRIPTION
**Pull request by vcfxb**
This PR includes several commits, but the most major change is that `Rc` has been replaced with `Arc` throughout much of the repository, in order to prepare for improvements from multithreading.

### CODE REVIEW

**Comment by alilleybrinker** [comment]
Looks like there are just some unused-related warnings, plus Conventional Commits issues, and then it's ready to merge. Probably just squash down to one commit and reword the message to match Conventional Commits.

**Comment by vcfxb** [comment]
@alilleybrinker assuming this last check is successful (it should be) I think this is ready to merge!

**Review by alilleybrinker — APPROVED** [review]

File: hipcheck/src/cli.rs [hunk 1]
```diff
@@ -388,6 +390,19 @@ pub enum Commands {
...
 	Ready,
+
+	/// Unstable command to aid in benchmarking
+	#[command(hide = true)]
+	Unstable(UnstableArgs),
+}
+
+// If no subcommand matched, default to use of '-t <TYPE> <TARGET' syntax. In
+// this case, `target` is a required field, but the existence of a subcommand
+// removes that requirement
```
Comment (j-lanson) [inline hipcheck/src/cli.rs:401]: If this comment was intentionally duplicated, is there a good reason for doing so?

File: hipcheck/src/unstable.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	cli::{CliConfig, UnstableArgs},
+	session::session::Check,
+	shell::{ColorChoice, Output},
+};
+use std::io::Write;
+use std::{io, time::Instant};
+use tempdir::TempDir;
+
+/// Handle unstable commands.
+pub fn main(args: UnstableArgs, config: &CliConfig) -> io::Result<()> {
```
Comment (j-lanson) [inline hipcheck/src/unstable.rs:11]: I would stylistically opt for a different name, such as `run` or `run_unstable`, but not a big deal

**Review by j-lanson — COMMENTED** [review]
I for one welcome our `Arc` overlords

File: hipcheck/src/cli.rs (hunk 1 above)
Comment (vcfxb) [reply hipcheck/src/cli.rs:401]: This was unintentional! Probably happened while merging. Thanks for catching this!

**Pull request merged**